Update Secrets Management Doc to contain Helm/Operator examples for automatically creating RBAC (#32820)
DataDog/documentation: content/en/agent/configuration/secrets-management.md

* Kubernetes Secrets: using `ENC[k8s_secret@namespace/secret-name/key]`

{{< tabs >}}
{{% tab "Datadog Operator" %}}
```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    secretBackend:
      command: "/readsecret_multiple_providers.sh"
```
{{% /tab %}}
{{% tab "Helm" %}}

> To use this executable with the Helm chart, set it as the following:

Author: Yes! Just added, thank you!

password: ENC[k8s_secret@database/database-secret/password]
```

> In this case, you must configure RBAC to allow the Agent's Service Account to read the Secret:

Reviewer: ```suggestion
Configure RBAC to allow the Agent's Service Account to read the Secret. The following Role grants read access to the `database-secret` Secret in the `database` namespace:
```

{{< tabs >}}
{{% tab "Helm" %}}
```yaml
datadog:
  (...)
  secretBackend:
    command: "/readsecret_multiple_providers.sh"
    roles:
      - namespace: database
        secrets:
          - database-secret
```
{{% /tab %}}
{{% tab "Operator" %}}
```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    secretBackend:
      command: "/readsecret_multiple_providers.sh"
      roles:
      - namespace: database
        secrets:
        - "database-secret"
```
{{% /tab %}}
{{< /tabs >}}

> In this example, a Role is created granting read access to the Secret `database-secret` in the `database` namespace.

Reviewer: ```suggestion
```

> You can also manually configure RBAC to allow the Agent's Service Account to read the Secret:

Reviewer: ```suggestion
Alternatively, you can define RBAC resources directly:
```
